notebook/serverextensions.py

```diff
@@ -0,0 +1,203 @@
+
+
+def toggle_serverextension_python(import_name, enabled=None, parent=None, user=False, sys_prefix=False):
+    """Toggle a server extension.
+    
+    Parameters
+    ----------
+    
+    import_name : str
+        Python import name of the extension
+    enabled : bool [default: None]
+        Toggle state for the extension.  Set to None to toggle, True to enable,
+        and False to disable the extension.
+    parent : Configurable [default: None]
+    user : bool [default: False]
+        Whether to install to the user's nbextensions directory.
+        Otherwise do a system-wide install (e.g. /usr/local/share/jupyter/nbextensions).
+    sys_prefix : bool [default: False]"""
+    config_dir = _get_config_dir(user=user, sys_prefix=sys_prefix)
+    cm = BaseJSONConfigManager(parent=parent, config_dir=config_dir)
+    cfg = cm.get("jupyter_notebook_config")
+    server_extensions = (
+        cfg.setdefault("NotebookApp", {})
+        .setdefault("nbserver_extensions", {})
+    )
+    if enabled:
+        server_extensions[import_name] = True
+    else:
+        if enabled is None:
+            if import_name not in server_extensions:
+                print("server extension not installed")
+            else:
+                server_extensions[import_name] = not server_extensions[import_name]
+        else:
+            server_extensions[import_name] = False
+    cm.update("jupyter_notebook_config", cfg)
+
+
+# ----------------------------------------------------------------------
+# Applications
+# ----------------------------------------------------------------------
+
+
+flags = {
+    "user" : ({
+        "ToggleServerExtensionApp" : {
+            "user" : True,
+        }}, "Install to the user's Jupyter directory"
+    ),
+    "sys-prefix" : ({
+        "ToggleServerExtensionApp" : {
+            "sys_prefix" : True,
+        }}, "Use sys.prefix as the prefix for installing nbextensions"
+    ),
+    "py" : ({
+        "ToggleServerExtensionApp" : {
+            "python" : True,
+        }}, "Install from a Python package"
+    ),
+}
+flags['python'] = flags['py']
+
+class ToggleServerExtensionApp(ToggleNBExtensionApp):
+
+    name = "jupyter serverextension enable/disable"
+    description = "Enable/disable a server extension using frontend configuration files."
+    
+    aliases = {}
+    flags = flags
+
+    user = Bool(False, config=True, help="Whether to do a user install")
+    sys_prefix = Bool(False, config=True, help="Use the sys.prefix as the prefix")
+    python = Bool(False, config=True, help="Install from a Python package")
+
+    def toggle_server_extension(self, import_name):
+        toggle_serverextension_python(import_name, self._toggle_value, parent=self, user=self.user, sys_prefix=self.sys_prefix)
+
+    def toggle_server_extension_python(self, package):
+        m, server_exts = _get_server_extension_metadata(package)
+        for server_ext in server_exts:
+            require = server_ext['require']
```

Comment:
'require' is not a very Pythonic name for this. Perhaps `module`? This is an importable module, right?